cpp/oneapi/dal/algo/knn/common_knn_dpc_test.cpp

```diff
@@ -0,0 +1,196 @@
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "gtest/gtest.h"
+#include "oneapi/dal/algo/knn/infer.hpp"
+#include "oneapi/dal/algo/knn/train.hpp"
+#include "oneapi/dal/table/row_accessor.hpp"
+
+using namespace oneapi;
+namespace nn = oneapi::dal::knn;
+
+TEST(knn_overflow_tests, classes_brute_force) {
```

Comment:
General comment for all tests cases: need better naming

Read this as an example: https://dzone.com/articles/7-popular-unit-test-naming

Possible naming in this case: throws_if_class_count_leads_to_overflow

Reply:
"Is it possible to have universal test cases for brutforce and kd-tree?" Do you mean that it's not necessary to test bad args for all methods? Also, Brute Force tests are expected to run on GPU while K-D tree ones - on CPU